### PR TITLE
feat: Allow `zeroIsNA` and `nNA` to process multiple assays

### PR DESCRIPTION
**Pull request by cvanderaa**
This PR solves [issue 50](https://github.com/rformassspectrometry/Features/issues/50). The main changes are: 
* `zeroIsNA` and `nNA` allow a vector of indices or names as argument `i`
* Added test units to cover the new features

### CODE REVIEW

File: R/Features-missing-data.R [hunk 1]
```diff
@@ -1,14 +1,25 @@
...
+  nNA <- sum(is.na(assay(x)))/prod(dim(x))
+  nNA_rows <- table(rowSums(is.na(assay(x))))
+  nNA_cols <- colSums(is.na(assay(x)))
+  return(list(nNA = nNA, nNArows = nNA_rows, nNAcols = nNA_cols))
+}
+
+## Internal wrapper function around `.nNA` for processing multiple assays
+## @param object A `Features` object
+## @param i One or more indices or names of the assay(s) to be processed. 
+.nNAi <- function(object, i){
```
Comment (lgatto) [inline R/Features-missing-data.R]: Always set a space between `)` and `{`.

File: R/Features-missing-data.R [hunk 2]
```diff
@@ -24,6 +35,33 @@
   k <= pNA
 }
 
+## Internal function for printing the result of nNA when applied to multiple
+## samples 
+## @param object A `Features` object
+## @param res A list of results obtained after applying `nNA` to multiple assays
+##     of `object`
+## @param i indices or names of the assays that were processed. 
+.printnNA <- function(object, res, i){
```
Comment (lgatto) [inline R/Features-missing-data.R]: This function is called *print* but I don't see any `print` or `cat` statements. Could you clarify.

File: R/Features-missing-data.R [hunk 3]
```diff
@@ -24,6 +35,33 @@
...
+  if(length(i) == 1) return(res[[1]])
```
Comment (lgatto) [inline R/Features-missing-data.R]: Please also use a space after `if`.

File: R/Features-missing-data.R [hunk 4]
```diff
@@ -85,16 +124,17 @@ setMethod("zeroIsNA", c("SummarizedExperiment", "missing"),
...
-              for (i in seq_len(length(object)))
-                  object[[i]] <- zeroIsNA(object[[i]])
-              object
+            for (i in seq_len(length(object)))
+              object[[i]] <- zeroIsNA(object[[i]])
+            object
           })
 
 ##' @rdname Features-missing-data
 setMethod("zeroIsNA", c("Features", "integer"),
```
Comment (lgatto) [inline R/Features-missing-data.R:124]: Do we need integer, or is numeric not enough?

**Review by lgatto — COMMENTED** [review]
See also comment in #50

File: R/Features-missing-data.R (hunk 4 above)
Comment (cvanderaa) [reply R/Features-missing-data.R:124]: I tested removing `setMethod("zeroIsNA", c("Features", "integer"){...})`, but this causes an infinite call between the generic method (from `AllGenerics.R`) and the method implementation (from `Features-mising-data.R`). See the traceback:
```
[...]
8: zeroIsNA(object, as.integer(i)) at AllGenerics.R#12
7: zeroIsNA(object, as.integer(i)) at Features-missing-data.R#133
6: zeroIsNA(object, as.integer(i)) at AllGenerics.R#12
5: zeroIsNA(object, as.integer(i)) at Features-missing-data.R#133
4: zeroIsNA(object, as.integer(i)) at AllGenerics.R#12
3: zeroIsNA(object, as.integer(i)) at Features-missing-data.R#133
2: zeroIsNA(ft0, 1) at AllGenerics.R#12
1: zeroIsNA(ft0, 1)
```
I think it is safer to leave it, unless you see an obvious fix.

**Comment by cvanderaa** [comment]
If you're ok with this update, you can merge it.

**Pull request merged**